actioneval/filesearch.go

```diff
@@ -0,0 +1,296 @@
+		result.Output.WriteString(fmt.Sprintf("%d\n", result.Lines))
+	}
+	return result
+}
+
+func (f *FileSearchFilter) satisfyGroupIdAnUserIdFilter(info os.FileInfo) bool {
+
+	var uid, gid uint32
+	//check groups
+	if info.Sys() != nil {
+
+		if sys, ok := info.Sys().(*syscall.Stat_t); ok {
+			uid = uint32(sys.Uid)
+			gid = uint32(sys.Gid)
+		}
+	}
+
+	if f.userId != -1 && uint32(f.userId) != uid {
+		return false
+	}
+
+	if f.groupId != -1 && uint32(f.groupId) != gid {
+		return false
+	}
+
+	return true
+}
+func (f *FileSearchFilter) satisfyAllFilters(info os.FileInfo) bool {
+
+	if f.filter != "" &&
+		!f.satisfyFilter(info.Name()) {
+		return false
+	}
+
+	// check if we satisfy the permission filter condition
+	if f.perm != 0 && f.sMode != 0 && !f.satisfyPermissionFilter(info) {
+		return false
+	}
+
+	// check if we satisfy the file type filter condition
+	if !f.satisfyFileType(info) {
+		return false
+	}
+
+	if !f.satisfyGroupIdAnUserIdFilter(info) {
+		return false
+	}
+
+	return true
+}
+
+func (f *FileSearchFilter) satisfyPermissionFilter(info os.FileInfo) bool {
+
+	filePerm := int64(info.Mode())
+	if (f.sMode == util.ModeExact && (filePerm&070000777 == f.perm)) ||
+		(f.sMode == util.ModeAnyBits && (filePerm&f.perm != 0)) ||
+		(f.sMode == util.ModeAllBits && (filePerm&f.perm == f.perm)) {
+		return true
+	}
+	return false
+}
+
+func (f *FileSearchFilter) satisfyFilter(filename string) bool {
+
+	if (f.filterType == util.TextFilterExact && strings.EqualFold(filename, f.filter)) ||
+		(f.filterType == util.TextFilterHasPrefix && strings.HasPrefix(filename, f.filter)) ||
+		(f.filterType == util.TextFilterHasSuffix && strings.HasSuffix(filename, f.filter)) ||
+		(f.filterType == util.TextFilterContains && strings.Contains(filename, f.filter)) {
+		return true
+	}
+	return false
+}
+
+//Verify the file type meets the criteria from yaml , i.e dir/symblink oor regular file
+func (f *FileSearchFilter) satisfyFileType(fileInfo os.FileInfo) bool {
+
+	fileInfo.Mode()
+	if (f.fileType == util.FileFilterDirectory && fileInfo.IsDir()) ||
+		(f.fileType == util.FileFilterSymblink && fileInfo.Mode()&os.ModeSymlink != 0) ||
+		(f.fileType == util.FileFilterRegularFile && fileInfo.Mode().IsRegular()) ||
+		(f.fileType == util.FileFilterAll) {
+		return true
+	}
+	return false
+}
+
+//The permission search mode concept has been taken from unix command "find -perm ",
+//which supports 3 modes, recognized by prefix '- or /'
+//where '-' prefix means all permission bits are set and  the '/' prefix  means any permissions bits are set.
+func parsePermission(perm string) (permInt int64, mode util.PermissionSearchMode, err error) {
+
+	if strings.HasPrefix(perm, "-") { // all permission bits are set for the file
+		mode = util.ModeAllBits
+	} else if strings.HasPrefix(perm, "/") { // any permissions are set for the file
+		mode = util.ModeAnyBits
+
+	} else {
+		mode = util.ModeExact
+	}
+	//strip non numeric chars
+	reg, err := regexp.Compile("\\D")
```

Comment:
Would be better to just compile the regex once (could use MustCompile). I can't help wondering if there isn't a better way to extract the numeric chars that you want here.